### PR TITLE
[13884] Use a single optionparser header for all tools and examples

### PR DESCRIPTION
**Pull request by MiguelBarro**
This implies distributing the third-party header as an interface table to allow the examples to be built standalone.
The version of the parser used is [this](https://raw.githubusercontent.com/HydrArgs/LeanAndMeanOptionParser/cpack/include/optionparser.h).
Windows installers have been successfully tested already.

### CODE REVIEW

**Comment by MiguelBarro** [comment]
@richiprosima Please be so gentle to re check style 💃

**Comment by MiguelBarro** [comment]
@richiprosima please check style

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
LGTM! I have just left a comment regarding the header guard. Let me know what you think. I have tested successfully this PR in Ubuntu Focal.

File: thirdparty/optionparser/optionparser.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file optionparser.hpp
+ *
+ */
+
+#ifndef OPTIONPARSER_HPP_
+#define OPTIONPARSER_HPP_
```
[the remark this answers is not in the document]
Comment (MiguelBarro) [reply thirdparty/optionparser/optionparser.hpp]: I miss the simplicity of `#pragma once`

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Pull request merged**